Source/santad/EventProviders/SNTDeviceManager.mm

```diff
@@ -199,43 +200,80 @@ - (void)handleAuthMount:(const es_message_t *)m
     return;
   }
 
-  long mountMode = m->event.mount.statfs->f_flags;
+  struct statfs *eventStatFS;
+  BOOL isRemount = NO;
+
+  switch (m->event_type) {
+    case  ES_EVENT_TYPE_AUTH_MOUNT:
+      eventStatFS = m->event.mount.statfs;
+      break;
+    case ES_EVENT_TYPE_AUTH_REMOUNT:
+      eventStatFS = m->event.remount.statfs;
+      isRemount = YES;
+      break;
+    default:
+      LOGE(@"Unexpected Event Type passed to DeviceManager handleAuthMount: %d", m->event_type);
+      // Fail closed.
+      es_respond_auth_result(self.client, m, ES_AUTH_RESULT_DENY, false);
+      return; 
+    break;
+  }
+
+  long mountMode = eventStatFS->f_flags;
   pid_t pid = audit_token_to_pid(m->process->audit_token);
   LOGI(@"SNTDeviceManager: mount syscall arriving from path: %s, pid: %d, fflags: %lu",
        m->process->executable->path.data, pid, mountMode);
 
   DADiskRef disk =
-    DADiskCreateFromBSDName(NULL, self.diskArbSession, m->event.mount.statfs->f_mntfromname);
+    DADiskCreateFromBSDName(NULL, self.diskArbSession, eventStatFS->f_mntfromname);
   CFAutorelease(disk);
 
   // TODO(tnek): Log all of the other attributes available in diskInfo into a structured log format.
   NSDictionary *diskInfo = CFBridgingRelease(DADiskCopyDescription(disk));
+  BOOL isInternal =  [diskInfo[(__bridge NSString *)kDADiskDescriptionDeviceInternalKey] boolValue];
   BOOL isRemovable = [diskInfo[(__bridge NSString *)kDADiskDescriptionMediaRemovableKey] boolValue];
-  BOOL isUSB =
-    [diskInfo[(__bridge NSString *)kDADiskDescriptionDeviceProtocolKey] isEqualTo:@"USB"];
-
-  if (!isRemovable || !isUSB) {
+  BOOL isEjectable = [diskInfo[(__bridge NSString *)kDADiskDescriptionMediaEjectableKey] boolValue];
+  NSString* protocol = diskInfo[(__bridge NSString *)kDADiskDescriptionDeviceProtocolKey];
```

Comment:
Do we know if other values besides USB and Virtual Interface are possible? 
Are we sure Virtual Interface means DMG in all cases?

Reply:
This gets set in the disk description via the IOKit registry in `kIOPropertyPhysicalInterconnectTypeKey` to`kIOPropertyPhysicalInterconnectTypeVirtual` for DMGs.  I've been testing manually with a few different DMGs. including the Santa-2022.5 release and they've all shown up as `Virtual Interface`.  

Searching through other tools and mailing lists is confirming this check. I'll make it tighter by checking that both `kDADiskDescriptionDeviceModelKey` is equal to "Disk Image" and `kDADiskDescriptionMediaNameKey` is equal to "disk image".  I'll move this check into a method.

Reply:
One issue I realized in testing the tighter check is that we'd break using RAM disks. The `Virtual Interface` check also covered ram disks. I've updated comments.

Reply:
Can we not break RAM disks? They can be useful and as I understand it they are not in scope for USB blocking.